packages/build/src/npm-packages/publish.ts

```diff
@@ -1,29 +1,40 @@
 import path from 'path';
-import { LERNA_BIN, PLACEHOLDER_VERSION, PROJECT_ROOT } from './constants';
+import {
+  EXCLUDE_RELEASE_PACKAGES,
+  LERNA_BIN,
+  MONGOSH_RELEASE_PACKAGES,
+  PROJECT_ROOT,
+} from './constants';
 import type { LernaPackageDescription } from './list';
 import { listNpmPackages as listNpmPackagesFn } from './list';
 import { spawnSync } from '../helpers/spawn-sync';
+import type { SpawnSyncOptionsWithStringEncoding } from 'child_process';
 
 export function publishNpmPackages(
-  isDryRun: boolean,
+  isDryRun = false,
+  isAuxiliaryOnly = false,
   listNpmPackages: typeof listNpmPackagesFn = listNpmPackagesFn,
   markBumpedFilesAsAssumeUnchangedFn: typeof markBumpedFilesAsAssumeUnchanged = markBumpedFilesAsAssumeUnchanged,
   spawnSyncFn: typeof spawnSync = spawnSync
 ): void {
-  const packages = listNpmPackages();
+  const commandOptions: SpawnSyncOptionsWithStringEncoding = {
+    stdio: 'inherit',
+    cwd: PROJECT_ROOT,
+    encoding: 'utf8',
+    env: {
+      ...process.env,
+      ...(isDryRun ? { npm_config_dry_run: 'true' } : {}),
+    },
+  };
+  let packages = listNpmPackages().filter(
+    (packageConfig) => !EXCLUDE_RELEASE_PACKAGES.includes(packageConfig.name)
+  );
 
-  const versions = Array.from(new Set(packages.map(({ version }) => version)));
-
-  if (versions.length !== 1) {
-    throw new Error(
-      `Refusing to publish packages with multiple versions: ${versions}`
+  if (isAuxiliaryOnly) {
+    packages = packages.filter(
+      (packageConfig) => !MONGOSH_RELEASE_PACKAGES.includes(packageConfig.name)
     );
   }
```

Comment:
Just double-checking, on a mongosh release we're still publishing the full list of packages, right?

How does this interact with the GH action scripts for publishing independent packages? Do we just need to be careful not to trigger them while a release is pending?

Reply:
Yeah I suppose that would be problematic if both happened at the same time. Perhaps if we move this to GitHub Actions we could also enforce that a `Release` action doesn't start if there's another `Release aux. packages` etc. one is running